### DISCUSSION TITLE
Wat zijn goede tools voor het testen op toegankelijkheid, voor web en app?

### DISCUSSION THREAD

**Discussion by rianrietveld** [nl-design-system/documentatie#740] [Toegankelijkheid]
Deel hier je tools. Wat gebruik je, wat is handig, waar gebruik je een bepaalde tool voor en hoe test je ermee?
Dan helpen we elkaar aan goede tools en ideeen om te testen op toegankelijkeheid.

**Comment by rianrietveld** [comment]
Mijn toolset:
- Mijn toetsenbord, duh.
- [Axe Devtools](https://www.deque.com/axe/browser-extensions/) in de browser, haalt al veel toegankelijkheidsfouten uit een gegenereerde webpagina.
- [HeadingsMap](https://rumoroso.bitbucket.io/) laat op een overzichtlijke manier de koppenstructuur van een webpagina zien.
- [Landmarks](https://matatk.agrip.org.uk/landmarks/) toont de landmarkstructuur van een webpagina.
- [text spacing bookmarklet](https://cdpn.io/stevef/debug/YLMqbo) van Steve Faulkner voor het testen van [WCAG succescriterium  1.3.12](https://www.w3.org/WAI/WCAG21/Understanding/text-spacing.html).
- [A11y Colour contrast table](https://rianrietveld.com/tools/contrast/) voor het testen van veel kleuren tegelijk.

**Comment by MeesoPost** [comment]
- [Colorblindly](https://github.com/oftheheadland/Colorblindly) om te kijken hoe een webpagina eruit ziet met verschillende types Kleurenblindheid. dit is er vaak eentje die ik vergeet dus om er een browser extensie voor te hebben help heel erg. 
- [ARC Toolkit](https://chromewebstore.google.com/detail/arc-toolkit/chdkkkccnlfncngelccgbgfmjebmkmce) Deze werkt vrijwel hetzelfde als [Axe Devtools](https://www.deque.com/axe/devtools/) maar geeft vaak nog andere resultaten, dus om deze te combineren is echt heel fijn.
- [Taba11y](https://chromewebstore.google.com/detail/taba11y/aocppmckdocdjkphmofnklcjhdidgmga)
 is een extensie voor chrome (en chromium based browsers)die de tab order visualiseert, erg handig om de flow van een website te testen

**Comment by jeffreylauwers** [comment]
- Naast de al genoemnde [Axe Devtools](https://www.deque.com/axe/browser-extensions/) soms ook de [WAVE Evaluation Tool-extensie](https://chromewebstore.google.com/detail/wave-evaluation-tool/jbbplnpkjmmeebjpijfedlgcdilocofh) voor Google Chrome.
- [Accessibility Scanner](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor&hl=en&gl=US&pli=1) van Android.
- Schermlezer van Apple: VoiceOver.
- Schermlezer van Android: Talkback.
- Toetsenbord.
- Tekst vergroten via instellingen browser.

En specifiek voor kleur...

- [Accessible Palette - Eugene Fedorenko](https://accessiblepalette.com/).
Online tool waarmee je een kleurenpalet kunt genereren en direct contrast kunt checken.
- [Contrast grid - Eightshapes](https://contrast-grid.eightshapes.com/)
Online tool waarbij je meerdere kleuren kunt invullen. Vervolgens ontstaat er een tabel waardoor je het contrast kunt checken.
- [Color review - Anton Robsarve](https://color.review/)
Online tool waarbij je twee kleuren kunt invullen. Vervolgens kun je checken hoeveel contrast deze kleurcombinatie oplevert. Daarnaast heb je de mogelijkheid om via een colorpicker naar voldoende contrast op zoek te gaan.
- [Who can use - Corey Ginnivan](https://whocanuse.com/)
Online tool waarbij je twee kleuren kunt invullen. Vervolgens kun je checken hoe deze kleurcombinatie overkomt bij verschillende visuele beperkingen.
- [Viz palette - Elijah Meeks & Susie Lu](https://projects.susielu.com/viz-palette)
Online tool waarbij je meerdere kleuren kunt invullen specifiek voor data visualisatie. Vervolgens kun je checken hoe deze kleurcombinatie overkomt bij verschillende visuele beperkingen.

**Comment by scar055** [comment]
[polypane](https://polypane.app/docs/) gebruik ik. Het is wel betaald, maar je krijgt als developer er veel uit want je krijgt dev tools. 

- de site te zien in 3 verschillende vensters (telefoon, tablet en desktop). 
- Kleurblindheid simulatie.
- visuele beperking simulatie als voorbeeld tunnel visie.
- verschillende helderheid van scherm als voorbeeld de zon schijnt er op
- en dyslexie simulatie
- color picker met ingebouwde contrast checker

**Comment by Wilco-acato** [comment]
Ik zie al hele goede tips en links naar nieuwe en voor mij al bekende tools. Super! 
Graag wil ik er nog een aan toevoegen die ik gebruik tijdens het ontwerpen, namelijk de Stark plugin voor Figma: https://www.getstark.co/figma/ 
Deze gebruik ik in Figma om op 'contrast', 'typografie', 'hierarchie' en 'touch-targets' te testen tijdens de designfase.

Voor developers is er ook een versie beschikbaar: https://www.getstark.co/for-developers/

**Comment by rianrietveld** [comment]
Focus-indicatorberekening van Robbert toevoegen als die af is, zie
https://github.com/nl-design-system/documentatie/issues/628#issuecomment-1976248748
Voorbeeld van berekening (work in progress!) van focus outline area. https://frameless.studio/design/focus-indicator

Dingen die nog ontbreken uit de berekening:

- achtergrondkleur container contrast ratio
- achtergrondkleur button contrast ratio
- border color contrast ratio
- box-shadow ring oppervlakte meetellen en contrast ratio berekenen
